benchmarks/xlmr/ootb/xlmr.py

```diff
@@ -76,40 +67,54 @@ def train(model, x_l, y_true_l, device=None, logger=None):
 
     #training loop
     learning_rate = 0.01
-    optimizer = torch.optim.SGD(model.parameters(), lr=learning_rate)
-    for x, y_true in zip(x_l, y_true_l):    
+    optimizer = torch.optim.SGD(xlmr.parameters(), lr=learning_rate)
+    for i, (x, y_true) in enumerate(zip(x_l, y_true_l)):   
         logger.batch_start()
         if device:
             x = x.to(device)
             y_true = y_true.to(device)
-        y_pred = model(x)
-        y_true = y_true.long()
-        loss = F.cross_entropy(y_pred[0], y_true[:,0,:]) # TODO: fix y_true data input hack
+        y_pred = xlmr.extract_features(x)
+        loss = F.cross_entropy(y_pred, y_true) 
         loss.backward()
         optimizer.step()
         optimizer.zero_grad() 
+
+        del y_pred
```

Comment:
This marks y_pred and loss as "deletable". Thus, the python garbage collector can go recover the memory, and the memory used during the computation of y_pred on gpu can also be recovered. If these aren't included then we see increasing memory usage as batches are run through the model, which shouldn't happen. Same thing in the inference function above.

Reply:
Do you want to add a comment to the code about this? Seems non-obvious

Reply:
Makes sense - I'll add a comment there. I was thinking if someone really needs to know they can look at git blame, but people may copy this code or just not look at git.